_targets.R

```diff
@@ -0,0 +1,50 @@
+library(targets)
+
+# Functions to fetch the raw data from Scibase (`fetch()`), process (and save) it for plotting (`process()`), 
+# and finally create a diagnostic log (`diagnostic_log()`) and associated figure (`line_range_plot()`).
+source(file = '1_fetch/src/fetch_fcn.R')
+source(file = '2_process/src/process_fcn.R')
+source(file = '2_process/src/write_csv_fcn.R')
+source(file = '3_visualize/src/line_range_plot_fcn.R')
+source(file = '3_visualize/src/diagnostic_log_fcn.R')
+
+tar_option_set(packages = c("tidyverse", "sbtools", "whisker"))
+
+list(
+  # Get the data from ScienceBase
+  tar_target(
+    model_RMSEs_csv,
+    fetch(sb_id = '5d925066e4b0c4f70d0d0599',
+          out_file = "1_fetch/out/model_RMSEs.csv"),
+    format = "file"
+  ), 
+  # Prepare the data for plotting
+  tar_target(
+    eval_data,
+    process(input_data = model_RMSEs_csv,
+            col_model_type = c('#1b9e77', '#d95f02', '#7570b3'), #pb, dl, pgdl, respectively,
+            pch_model_type = c(21, 22, 23)) #pb, dl, pgdl, respectively),
+  ),
+  # Create a plot
+  tar_target(
```

Comment:
These last three targets - `figure_1_png`, `model_summary_results_csv`, and `model_diagnostic_text_txt` - are all creating files. Much like your first target does, we need to tell `targets` that these are files and it should track the _contents_ of the files rather than the file name. To set them up as file targets, you need to 1) add `format = "file"` to your `tar_target()` call for each, and 2) make sure that all the functions being used have `return(file_out)` (where `file_out` is whatever the object is in your function that has the filepath character string).

Reply:
Yes, it needs both to work as expected! If you have `format='file'` but the function doesn't return the filepath, I believe you get a mysterious error when you try to build. Alternatively, if you have the filepath return but are missing `format="file"`, then `targets` just looks at the filepath as if it were any character string and will not consider that target outdated unless the filename (and not the contents of that file) changes.